### DISCUSSION TITLE
Webhook on local telegram bot server without domain field

### DISCUSSION THREAD

**Discussion by Peterculazh** [telegraf/telegraf#1972] [Q&A]
Hello, I've started local telegram bot server and managed with using NestJS connect webhook to local telegram server via http requests and start getting responses. Decided switch to telegraf because I'm not enjoying getting all requests in single endpoint and there in options for setting up webhook comes a domain as required field. Since I'm doing that locally and I haven't a domain, how do I can implement setting up webhook with local telegram server?
Thanks :)

**Comment by MKRhere** [comment]
`bot.launch` always adds https to the domain; so what you could do is, listen for updates with a webserver, and call `bot.handleUpdate(update)` with the JSON object received from the API. Then send a request to the local Bot API with something like:

```shell
curl "http://localhost:8081/bot$TOKEN/setWebhook?url=http://localhost:$BOT_PORT/webhook"
```
